site/content/blog/ytt-validations-preview.md

```diff
@@ -0,0 +1,124 @@
+
+So, `ytt` Validations is the first of these such experiments.
+
+We're doing this so that you have a voice in making sure features like this meet _your_ needs.
+
+### How do I get started?
+
+`ytt` v0.41.0 will be released soon. When it does, grab yourself the latest.
+
+Then, to enable Validations, and start including validation rules in a copy of _your_ schema:
+
+```yaml
+#@data/values-schema
+---
+dex:
+  #@schema/validation ("non-empty value", lambda v: len(v) > 1)
+  username: ""
+  #@schema/validation ("non-empty value", lambda v: len(v) > 1), ("not 'default'", lambda v: v != "default")
+  namespace: ""
+  #@schema/validation ("one of: aws, azure, vsphere", lambda v: v in ["aws", "azure", "vsphere"])
+  provider: ""
+```
+
+... and set the OS environment variable, `YTTEXPERIMENTS`...
+
+```console
+$ YTTEXPERIMENTS=validations yttt -f schema.yml -v dex.namespace=default
+ytt: Error: One or more data values were invalid:
+- "username" (schema.yml:5) requires "non-empty value" (by schema.yml:4)
+- "namespace" ((data-value arg):1) requires "not 'default'" (by schema.yml:6)
+- "provider" (schema.yml:9) requires "one of: aws, azure, vsphere" (by schema.yml:8)
+```
+
+You'll notice that with the validations experiment turned off, `ytt` doesn't recognize those annotations.
+
+```console
+$  yttt -f schema.yml -v dex.namespace=default --data-values-inspect
+dex:
+  username: ""
+  namespace: default
+  provider: ""
+  ```
+
+### What is implemented so far?
+
+As of v0.41.0, we've implemented the core behavior of recognizing, parsing, and checking validations in schema.
+
+In essence, we've introduced the `@schema/validation` annotation and wired it in to [the `ytt` pipeline](../ytt/docs/v0.40.0/how-it-works/) — specifically, at the end of the "Calculate Data Values" step.
+
+We're including light documentation as we go, so check out `[link to @schema/validation in schema reference]` for the exact details of what's available.
+
+### How can I give feedback?
+
+We would love to tell you, thank you for asking!
+
+Delightfully, any format that is most convenient for you, we're happy to hear your thoughts!
+
+The likely easiest/best way is by popping by our channel on the Kubernetes Slack workspace: [#carvel](https://kubernetes.slack.com/archives/CH8KCCKA5). You can fork a thread, there; request a direct message conversation; or even a video call. If you don't yet have an account there, you can get an invite at http://slack.k8s.io/.
+
+That said, if you prefer the long-form, feel free to [create a GitHub Issue in our repo](https://github.com/vmware-tanzu/carvel-ytt/issues/new?assignees=&labels=carvel+triage&template=other-issue.md&title=Feedback+for+ytt+Validations).
+
+
+### Why are Validations Useful?
+
+To date, ytt Library / Carvel Package authors have had to "roll their own" logic to check that customer's inputs are valid: that data values are present/non-empty and within acceptable values.
```

Comment:
customer or consumer?

Reply:
yeah... I thought "customer" would be a little more appropriate because it's about the relationship between the Package Author and the people using their stuff.... 

Thoughts?